app/interface/clients/barun_client.py

```
@@ -0,0 +1,56 @@
+            headers=HEADERS,
+            timeout=self._timeout,
+        )
+
+        try:
+            async with self._session.post(
+                BARUN_LOGIN_URL,
+                data={"email": self.student_id, "password": self.password},
+                allow_redirects=True,
+            ) as login_response:
+                if login_response.status != 200:
+                    raise ClientException(error_type=ErrorType.BARUN_ERROR)
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)